components/canvas/raqote_backend.rs

```diff
@@ -244,16 +247,30 @@ pub fn source<'a>(pattern: &Pattern<'a>) -> raqote::Source<'a> {
             pattern.radius2,
             raqote::Spread::Pad,
         ),
-        Pattern::Surface(pattern) => raqote::Source::Image(
-            pattern.image,
-            pattern.extend,
-            pattern.filter,
-            pattern.transform,
-        ),
-    }
-}
-
-impl PatternHelpers for Pattern<'_> {
+        Pattern::Surface(pattern) => {
+            #[allow(unsafe_code)]
+            let data = unsafe {
+                let data = pattern.image.as_raw_bytes();
+                std::slice::from_raw_parts(
+                    data.as_ptr() as *const u32,
+                    data.len() / std::mem::size_of::<u32>(),
+                )
```

Comment:
It's unfortunate that the `raqote` API requires `unsafe` here.

Reply:
Indeed. Hopefully vello_cpu will be able to replace raqote.